src/hooks/useResponsive.ts

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from 'react'
+import { matchMediaQuery as defaultMatchMediaQuery } from '@styles'
+
+type Device = 'desktop' | 'tablet' | 'mobile'
+type MaxWidth = number
+
+type ResponsiveCase<T, K extends keyof T> = {
+  [key in Device]: T[K]
+}
+type MatchMediaQuery = {
+  [key in Device]: `(max-width: ${MaxWidth}px)`
+}
+type MatchMediaQueryParams = MatchMediaQuery | undefined
+
+/**
+ *
+ * @example
+ * const avatarSize = useResponsive<AvatarProps, 'size'>({
+ *   desktop: 'medium',
+ *   tablet: 'small',
+ *   mobile: 'small'
+ * });
+ */
+export const useResponsive = <T, K extends keyof T>(
+  responsiveCase: ResponsiveCase<T, K>,
+  matchMediaQuery: MatchMediaQueryParams = defaultMatchMediaQuery as MatchMediaQuery
+): T[K] => {
+  const getDevice = (queries: MatchMediaQuery): Device => {
+    const defaultDevice: Device = 'desktop'
+
+    if (typeof window === 'undefined') {
+      return defaultDevice
+    }
+
+    return Object.keys(queries).reduce<Device>((prev, cur) => {
+      const device = cur as Device
+      const query = queries[device] as string
+
+      return window.matchMedia(query).matches ? device : prev
+    }, defaultDevice)
+  }
+
+  const [currentDevice, setCurrentDevice] = useState<Device>(
+    getDevice(matchMediaQuery)
+  )
+
+  useEffect(() => {
+    const matchMedia = Object.keys(matchMediaQuery).map(device => {
+      const query = matchMediaQuery[device as Device] as string
+      return window.matchMedia(query)
+    })
```

Comment:
위와 동일하게 요렇게 수정하면 불필요한 string 단언을 제거할 수 있을 것 같아유 !!

```suggestion
     const matchMedia = (Object.keys(matchMediaQuery) as Device[]).map(
      device => {
        const query = matchMediaQuery[device]
        return window.matchMedia(query)
      }
    )
```

Reply:
[6417182](https://github.com/price-offer/offer-fe/pull/53/commits/6417182f5f0d81713dd0dd33c593383f623f527b)에서 적용하였습니다! 감사합니다!